File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    const dates = new Array(speeds.length);
    for (let i = 0; i < speeds.length; i++) {
        dates[i] = Math.ceil((100 - progresses[i]) / speeds[i]);
    }
    
    console.log(dates);
    
    const result = [];
    let max = dates[0];
    let count = 0;
    let i;
    for (i = 0; i < dates.length; i++) {
        if (dates[i] <= max) {
            count++;
        } else {
            result.push(count);
            max = dates[i];
            i--;
            count = 0;
        }
    }
    result.push(count);
    
    return result;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
from collections import deque

def solution(progresses, speeds):
    result = []
    queue = deque(progresses)
    speed_queue = deque(speeds)
    
    while queue:
        count = 0
        for i in range(len(queue)):
            queue[i] += speed_queue[i]
        
        while queue and queue[0] >= 100:
            queue.popleft()
            speed_queue.popleft()
            count += 1
        
        if count != 0:
            result.append(count)
        
    return result
        